datasetinsights/stats/visualization/keypoints_pose.py

```diff
@@ -0,0 +1,133 @@
+import math
+from typing import Any, Dict, List, Tuple
+
+import numpy as np
+
+from datasetinsights.stats.visualization.constants import (
+    COCO_KEYPOINTS,
+    COCO_SKELETON,
+)
+
+
+def _is_torso_visible_or_labeled(kp: List) -> bool:
+    """
+    True if torso (left hip, right hip, left shoulder,
+    right shoulder) is visible else False
+    """
+    return (
+        (kp[17] == 1 or kp[17] == 2)
+        and (kp[20] == 1 or kp[20] == 2)
+        and (kp[41] == 1 or kp[41] == 2)
+        and (kp[38] == 1 or kp[38] == 2)
+    )
+
+
+def _get_kp_where_torso_visible(annotations: List) -> List:
+    """
+    List of keypoint where torso is visible or labeled
+    """
+    keypoints = []
+    for ann in annotations:
+        if _is_torso_visible_or_labeled(ann):
+            keypoints.append(ann)
+    return keypoints
+
+
+def _calc_mid(p1: Tuple[Any, Any], p2: Tuple[Any, Any]):
+    """
+    Calculate mid point of two points
+    """
+    return (p1[0] + p2[0]) / 2, (p1[1] + p2[1]) / 2
+
+
+def _calc_dist(p1: Tuple[Any, Any], p2: Tuple[Any, Any]) -> float:
+    """
+    Calculate distance between two points
+    """
+    return math.sqrt(((p1[0] - p2[0]) ** 2) + ((p1[1] - p2[1]) ** 2))
+
+
+def _translate_and_scale_xy(X: List, Y: List):
```

Comment:
Can you use numpy array instead of lists?

Reply:
Is there any reason for using numpy array here?

Reply:
yes for computation only